addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -452,6 +456,10 @@ def _process_accrual_plans(self, date_to=False, force_period=False, log=True):
             # get current level and normal period boundaries, then set nextcall, adjusted for level transition and carryover
             # add days, trimmed if there is a maximum_leave
             while allocation.nextcall <= date_to:
+                if allocation.holiday_status_id.request_unit in ["day", "half_day"]:
+                    leaves_taken = _get_leaves_taken(allocation)
+                else:
+                    leaves_taken = _get_leaves_taken(allocation) / allocation.employee_id._get_hours_per_day(allocation.date_from)
```

Comment:
while take `allocation.date_from` and not `allocation.nextcall`? That should give a more accurate value